src/python/pants/backend/python/goals/coverage_py.py

```diff
@@ -113,112 +120,87 @@ class CoverageSubsystem(PythonToolBase):
-            ),
-        )
-        register(
-            "--config-discovery",
-            type=bool,
-            default=True,
-            advanced=True,
-            help=(
-                "If true, Pants will include any relevant config files during runs "
-                "(`.coveragerc`, `setup.cfg`, `tox.ini`, and `pyproject.toml`)."
-                f"\n\nUse `[{cls.options_scope}].config` instead if your config is in a "
-                f"non-standard location."
-            ),
-        )
-        register(
-            "--global-report",
-            type=bool,
-            default=False,
-            help=(
-                "If true, Pants will generate a global coverage report.\n\nThe global report will "
-                "include all Python source files in the workspace and not just those depended on "
-                "by the tests that were run."
-            ),
-        )
-        register(
-            "--fail-under",
-            type=float,
-            default=None,
-            help=(
-                "Fail if the total combined coverage percentage for all tests is less than this "
-                "number.\n\nUse this instead of setting fail_under in a coverage.py config file, "
-                "as the config will apply to each test separately, while you typically want this "
-                "to apply to the combined coverage for all tests run."
-                "\n\nNote that you must generate at least one (non-raw) coverage report for this "
-                "check to trigger.\n\nNote also that if you specify a non-integral value, you must "
-                "also set [report] precision properly in the coverage.py config file to make use "
-                "of the decimal places. See https://coverage.readthedocs.io/en/latest/config.html ."
-            ),
-        )
-
-    @property
-    def filter(self) -> tuple[str, ...]:
-        return tuple(self.options.filter)
-
-    @property
-    def reports(self) -> tuple[CoverageReportType, ...]:
-        return tuple(self.options.report)
+    filter = StrListOption(
+        "--filter",
+        help=(
+            "A list of Python modules or filesystem paths to use in the coverage report, e.g. "
+            "`['helloworld_test', 'helloworld/util/dirutil'].\n\nBoth modules and directory "
+            "paths are recursive: any submodules or child paths, respectively, will be "
+            "included.\n\nIf you leave this off, the coverage report will include every file "
+            "in the transitive closure of the address/file arguments; for example, `test ::` "
+            "will include every Python file in your project, whereas "
+            "`test project/app_test.py` will include `app_test.py` and any of its transitive "
+            "dependencies."
+        ),
+    )
+    reports = EnumListOption(
+        "--report",
+        default=[CoverageReportType.CONSOLE],
+        help="Which coverage report type(s) to emit.",
+    )
+    _output_dir = StrOption(
```

Comment:
Odd this wasn't a dir option :thinking:

Reply:
Intentionally so. `dir_option` checks that the dir already exists and errors if not. That's not desirable here.